daemon/create.go

```diff
@@ -142,13 +142,40 @@ func (daemon *Daemon) generateSecurityOpt(ipcMode containertypes.IpcMode, pidMod
 	if ipcMode.IsHost() || pidMode.IsHost() {
 		return label.DisableSecOpt(), nil
 	}
-	if ipcContainer := ipcMode.Container(); ipcContainer != "" {
+
+	var ipcLabel []string
+	var pidLabel []string
+	ipcContainer := ipcMode.Container()
+	pidContainer := pidMode.Container()
+	if ipcContainer != "" {
 		c, err := daemon.GetContainer(ipcContainer)
 		if err != nil {
 			return nil, err
 		}
+		ipcLabel = label.DupSecOpt(c.ProcessLabel)
+		if pidContainer == "" {
+			return ipcLabel, err
+		}
+	}
+	if pidContainer != "" {
+		c, err := daemon.GetContainer(pidContainer)
+		if err != nil {
+			return nil, err
+		}
 
-		return label.DupSecOpt(c.ProcessLabel), nil
+		pidLabel = label.DupSecOpt(c.ProcessLabel)
+		if ipcContainer == "" {
+			return pidLabel, err
+		}
+	}
+
+	if pidLabel != nil && ipcLabel != nil {
+		for i := 0; i < len(pidLabel); i++ {
+			if pidLabel[i] != ipcLabel[i] {
```

Comment:
Ok, LGTM then! (IANAM)

On Fri, May 13, 2016, 8:33 PM Mrunal Patel <notifications@github.com> wrote:

> In daemon/create.go
> <https://github.com/docker/docker/pull/22481#discussion_r63270556>:
>
> >
> > -		return label.DupSecOpt(c.ProcessLabel), nil
> > +		pidLabel = label.DupSecOpt(c.ProcessLabel)
> > +		if ipcContainer == "" {
> > +			return pidLabel, err
> > +		}
> > +	}
> > +
> > +	if pidLabel != nil && ipcLabel != nil {
> > +		for i := 0; i < len(pidLabel); i++ {
> > +			if pidLabel[i] != ipcLabel[i] {
>
> @mlaventure <https://github.com/mlaventure> No, that can't happen.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/docker/docker/pull/22481/files/4772a40138e8f88e77f04049c1c2dc4865508dc7#r63270556>
>